Rover/sailboat.cpp

```diff
@@ -552,3 +575,113 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+void Sailboat::update_in_irons()
+{
+    // only enabled for separate mainsail and options bit set
+    {
+        const bool separate_mainsail_channel =
+            (channel_mainsail !=nullptr) &&
+            (channel_mainsail != rover.channel_throttle);
+
+        if (!(separate_mainsail_channel &&
+              option_is_set(OPTION::GET_OUT_OF_IRONS_IN_AUTO))) {
+            clear_in_irons();
+            return ;
+        }
+    }
+
+    const float wind_dir_apparent =
+        degrees(rover.g2.windvane.get_apparent_wind_direction_rad());
+
+    // Already "in irons" or not?
+    if (!_in_irons.state) {
+
+        // check conditions to enter "in irons limbo"
+
+        // being blown backwards by the wind
+        bool  going_backwards = false;
+        {
+            float speed;
+            // only useful if we have a valid speed (e.g fails if no gps fix)
+            if (rover.g2.attitude_control.get_forward_speed(speed)) {
+                going_backwards = is_negative(speed);
+            }
+        }
+
+        // only pure sailboats get into irons
+        const bool cannot_motorsail = this->motor_state == UseMotor::USE_MOTOR_NEVER;
+
+        // only in irons if can't get power from sails
+        const bool headed = abs(wind_dir_apparent) < sail_no_go;
+
+        if (cannot_motorsail && going_backwards && headed) {
+            // in_irons_limbo_start_ms == 0  means not yet "in irons limbo"
+            if (_in_irons.start_ms == 0) {
+                // start "in irons limbo"
+                _in_irons.start_ms = millis();
+            } else {
+                // signs of signal to the rudder to rotate required direction
+                constexpr int rotate_clockwise = -1;
+                constexpr int rotate_anticlockwise = 1;
+
+                // we have been in the "in irons limbo" period for a while
+                if ((millis() - _in_irons.start_ms) >= IN_IRONS_LIMBO_MS) {
+
+                    // OK, long enough
+
+                    // fix the desired heading while in irons
+                    _in_irons.heading_rad =
+                        (currently_tacking)
+                        ? tack_heading_rad
+                        : radians(rover.g2.wp_nav.oa_wp_bearing_cd() * 0.01)
+                        ;
```

Comment:
```suggestion
                    _in_irons.heading_rad = currently_tacking ? tack_heading_rad :  radians(rover.g2.wp_nav.oa_wp_bearing_cd() * 0.01);
```
It is a bit upsetting that wpnav does not provide a radians value. If you want to use 5 lines you might as well do a full if statement ;)

Reply:
Couldn't the wp nav bearing be in the no-go zone?